### ISSUE TITLE
error java.lang.IllegalArgumentException: Receiver not registered: com.centaurwarchief.smslistener.SmsReceiver@b16a902

### ISSUE THREAD

**Issue by duranmla** [CentaurWarchief/react-native-android-sms-listener#4]
# Description

This error happens when the application **goes to background using the back button of the system**, if I put the application on background by hitting "home" or "see background application button" this error is not triggered.

What I have done by now is to set a try catch on the unregister method, even when it avoid the application to crash it doesn't solve the right problem.

![screenshot_6_26_16__7_15_pm](https://cloud.githubusercontent.com/assets/1425162/16365876/778b8e3c-3bd2-11e6-818b-1a80e1906100.jpg)

snippet to avoid this:

``` java
// line 42 of SmsListenerModule.java
private void unregisterReceiver(BroadcastReceiver receiver) {
        try {
            mActivity.unregisterReceiver(receiver);
            mReceiver = null;
        } catch (IllegalArgumentException e) {
            e.printStackTrace();
        }
    }
```
# Version of module

0.2.0
# Device

HTC M8 - OS 6.0
# Error

![messagedemographic_java_-_wobe_-____mobile_client_wobe_](https://cloud.githubusercontent.com/assets/1425162/16365900/509f17fc-3bd3-11e6-81ba-e30979ae7ff9.jpg)

**Comment by duranmla** [comment]
Seems like this issue has been solved already on your code guys, but it doesn't have a reference of it on NPM

**Comment by andreyvital** [comment]
@duranmla just published 2.1.0 ─ it should work now. Thanks for reporting this.
